src/main/java/eu/merloteducation/serviceofferingorchestrator/service/ServiceOfferingsService.java

```diff
@@ -382,11 +380,17 @@ public SelfDescriptionMeta addServiceOffering(MerlotServiceOfferingCredentialSub
             }
         }
 
-        patchTermsAndConditions(credentialSubject);
+        MerlotParticipantDto participantDto = organizationOrchestratorClient
+            .getOrganizationDetails(credentialSubject.getOfferedBy().getId());
 
+        TermsAndConditions termsAndConditions = ((MerlotOrganizationCredentialSubject) participantDto.getSelfDescription()
+            .getVerifiableCredential().getCredentialSubject()).getTermsAndConditions();
+        patchTermsAndConditions(credentialSubject, termsAndConditions);
+
+        OrganisationSignerConfigDto orgaSignerConfig = participantDto.getMetadata().getOrganisationSignerConfigDto();
```

Comment:
maybe add a small error handler if the verification method and/or private key are not set